# HG changeset patch
# User Carlo de Falco <kingcrimson@tiscali.it>
# Date 1321137951 -3600
#      Sat Nov 12 23:45:51 2011 +0100
# Node ID 4d8317915db9224291001f389e4960caa71de2d6
# Parent  9326fc1f0c71658367aecc8a47a9e894f8d047e3
fix extra output parameters in gmres

* gmres.m: fix behaviour of output parameters iter and resvec

diff --git a/scripts/sparse/gmres.m b/scripts/sparse/gmres.m
--- a/scripts/sparse/gmres.m
+++ b/scripts/sparse/gmres.m
@@ -60,17 +60,17 @@
 ## total iterations performed.
 ## @item @var{resvec} is a vector containing the relative residual at each
 ## iteration.
 ## @end itemize
 ##
 ## @seealso{bicg, bicgstab, cgs, pcg}
 ## @end deftypefn
 
-function [x, flag, presn, it] = gmres (A, b, restart, rtol, maxit, M1, M2, x0)
+function [x, flag, presn, it, resids] = gmres (A, b, restart, rtol, maxit, M1, M2, x0)
 
   if (nargin < 2 || nargin > 8)
     print_usage ();
   endif
 
   if (ischar (A))
     Ax = str2func (A);
   elseif (ismatrix (A))
@@ -177,17 +177,17 @@ function [x, flag, presn, it] = gmres (A
     iter++;
   endwhile
 
   if (presn > rtol * prec_b_norm)
     flag = 0;
   endif
 
   resids = resids(1:iter-1);
-  it = [floor(maxit/restart), rem(maxit, restart)];
+  it = [ceil(iter / restart), rem(iter, restart)];
 
 endfunction
 
 
 %!shared A, b, dim
 %! dim = 100;
 %!test
 %! A = spdiags ([-ones(dim,1) 2*ones(dim,1) ones(dim,1)], [-1:1], dim, dim);
